src/ml_board/frontend/Dashboard/src/worker_socket/WorkerSocket.ts

```diff
@@ -0,0 +1,107 @@
+import socketIO, { Socket } from 'socket.io-client';
+import { settingConfigsInterface } from '../app/App';
+import { DataFromSocket } from './DataTypes';
+import { connectionMTCb, msgCounterIncMTCb, pingMTCb, throughputMTCb, updateMTCb } from './MainThreadCallbacks';
+
+
+// ========================= variables ============================//
+
+// const DEFAULT_URL = 'http://localhost:7000/'; // or http://127.0.0.1:7000/'
+
+let interval: NodeJS.Timer;
+let lastPing: number = -1;
+let lastPong: number = -1;
+const period: number = 10;
+let msgCountPerPeriod: number = 0;
+let socket: Socket;
+
+
+// =~=~=~=~=~=~=~=~=~=~=~=~=~= ~WebSocket~ =~=~=~=~=~=~=~=~=~=~=~=~=~=~=~=//
+const initSocket = (settingConfigs: settingConfigsInterface) => {
+    socket = socketIO(settingConfigs.socketConnectionUrl, { autoConnect: true });
+    console.log("WebSocket initialized");
+    socket.on('connect', () => connect(socket, settingConfigs.gridSearchId));
+    socket.on('disconnect', disconnect);
+    socket.on('connect_error', connect_error);
+    socket.on('mlgym_event', mlgym_event);
+    socket.on('pong', pong);
+};
+
+
+// ========================= connection events ============================//
+const connect = (socket: Socket, runId: string) => {
+    // TODO:ASK how exactly should the join happen? and this was in the old code const runId = "mlgym_event_subscribers";
+    // Max added bug report here: https://github.com/mlgym/mlgym/issues/134
+    socket.emit('join', { rooms: [runId] });
+    // start periodic server pining
+    interval = setInterval(pinging, period * 1000, socket);
+    // flag main thread that connection is on
+    connectionMTCb(true);
+};
+
+const disconnect = (reason: Socket.DisconnectReason) => stop(reason);
+
+const connect_error = (err: Error) => stop(err);
+
+
+// ========================= data driven events ============================//
+// const mlgym_event = (msg:JSON) => {
+const mlgym_event = (msg: string) => {
+    const parsedMsg: DataFromSocket = JSON.parse(msg);
```

Comment:
Not entirely sure if we need to parse this with the proper MLgym backend. We should fix the DummyServer to always send proper JSON instead of strings. see  #149

Reply:
we can have a new PR for it based on the issue  :)